realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -61,7 +66,8 @@ public Table() {
         if (nativePtr == 0) {
             throw new java.lang.OutOfMemoryError("Out of native memory.");
         }
-        sharedRealm = null;
+        this.sharedRealm = null;
+        context.addReference(this);
```

Comment:
The Context is created while the root component is created. This constructor is mainly to support the case that a Table can be created without any Group or SharedGroup. Not a real case but only for testing purpose. We can remove this constructor in the future. So, the rule is like when your table is created from SharedRealm, the context is created by the SharedRealm since that is the root component and everything branched out from that should be synced with that context when it get destructed. For the Table created from nowhere, like here, the Table itself it he root component.